graylog2-web-interface/src/views/components/aggregationwizard/aggregationElements/GroupByElement.ts

```diff
@@ -14,16 +14,190 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
+import AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import Pivot, { TimeConfigType, ValuesConfigType } from 'views/logic/aggregationbuilder/Pivot';
+
 import type { AggregationElement } from './AggregationElementType';
 
+import type { GroupingDirection, DateGrouping, ValuesGrouping, GroupByFormValues, WidgetConfigFormValues } from '../WidgetConfigForm';
 import GroupByConfiguration from '../elementConfiguration/GroupByConfiguration';
 
+type GroupByError = {
+  field?: string,
+  limit?: string,
+  interval?: string,
+};
+
+const validateDateGrouping = (grouping: DateGrouping): GroupByError => {
+  const groupByError = {} as GroupByError;
+
+  if (!grouping.field?.field) {
+    groupByError.field = 'Field is required.';
+  }
+
+  if (grouping.interval.type === 'auto') {
+    if (!grouping.interval.scaling) {
+      groupByError.interval = 'Scaling is required.';
+    }
+
+    if (grouping.interval.scaling && (grouping.interval.scaling <= 0 || grouping.interval.scaling >= 10)) {
+      groupByError.interval = 'Must be greater than 0 and smaller or equals 10.';
+    }
+  }
+
+  if (grouping.interval.type === 'timeunit') {
+    if (!grouping.interval.value) {
+      groupByError.interval = 'Value is required.';
+    }
+
+    if (grouping.interval.value && grouping.interval.value <= 0) {
+      groupByError.interval = 'Must be greater than 0.';
+    }
+  }
+
+  return groupByError;
+};
+
+const validateValuesGrouping = (grouping: ValuesGrouping): GroupByError => {
+  const groupByError = {} as GroupByError;
```

Comment:
We can also do:

```suggestion
  const groupByError: GroupByError = {};
```
It does not make any difference here, just uses less force to assert the type. `as` should be used as a last resort.